dataSourceImpl/src/main/java/com/panhuk/dataSourceimpl/leaderboard/LeaderboardDSImpl.kt

```diff
@@ -0,0 +1,21 @@
+package com.panhuk.dataSourceimpl.leaderboard
+
+import com.panhuk.database.LeaderboardDao
+import com.panhuk.datasource.LeaderboardDS
+import com.panhuk.domain.model.Leaderboard
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class LeaderboardDSImpl(private val leaderboardDao: LeaderboardDao) : LeaderboardDS {
+  override suspend fun insert(leaderboard: Leaderboard) {
+    leaderboardDao.insert(leaderboard.mapToDB())
```

Comment:
it might be better to do all the mapping beforehand, so that the insert method does only insertion

Reply:
consider changing param type to the DB type so that insert method does only the insertion and mapping is done somewhere else

Reply:
override suspend fun insert(leaderboard: Leaderboard) {
    leaderboardDao.insert(leaderboard.mapToDB())
    val databaseModel = leaderboard.mapToDB()
    leaderboardDao.insert(databaseModel)
  }

Reply:
Wait a moment. I changed the code to this. Maybe there is some mistake with git ?

Reply:
no mistake, rather misunderstanding

there's still mapping done inside the insert method, how about changing the param leaderboard from Leaderboard type to LeaderboardDB type? this way we won't break the S principle

Reply:
So you want me to set a two mappers. When we get the data from DB we will map it in data source . And when we want to transform data from domain -> DB we will use mapper in Repository layer so in datasource we will get our DB model ?